st2api/setup.py

```diff
@@ -14,27 +14,43 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-try:
-    from setuptools import setup, find_packages
-except ImportError:
-    from ez_setup import use_setuptools
-    use_setuptools()
-    from setuptools import setup, find_packages
+import re
+import os.path
+from pip.req import parse_requirements
+from setuptools import setup, find_packages
+
+
+def fetch_requirements():
+    links = []
+    reqs = []
+    for req in parse_requirements('requirements.txt', session=False):
+        if req.link:
+            links.append(str(req.link))
+        reqs.append(str(req.req))
+    return (reqs, links)
+
+
+current_dir = os.path.dirname(os.path.realpath(__file__))
+version_file = os.path.join(current_dir, '../st2client/st2client/__init__.py')
+with open(version_file, 'r') as f:
+    vmatch = re.search(r'__version__ = [\'\"](.*)[\'\"]$', f.read(), flags=re.MULTILINE)
+
+
+install_reqs, dep_links = fetch_requirements()
+ST2_COMPONENT = os.path.basename(current_dir)
+ST2_VERSION = vmatch.group(1)
+
 
 setup(
-    name='st2api',
-    version='0.4.0',
-    description='',
+    name=ST2_COMPONENT,
+    version=ST2_VERSION,
+    description='{} component'.format(ST2_COMPONENT),
     author='StackStorm',
     author_email='info@stackstorm.com',
-    install_requires=[
-        "pecan",
-    ],
-    package_data={
```

Comment:
I've removed that since, install_requires is not needed as we can use 
load of requirements using pip. And if you mean package_data, since 
we've got no exact and working way to select need files I started to 
use MANIFEST.in where simply all files from the source directory are 
included.

On Mon, Jun 22, 2015 at 2:46 PM, Tomaz Muraus 
<notifications@github.com> wrote:
> In st2api/setup.py:
> 
> >      author='StackStorm',
> >      author_email='info@stackstorm.com',
> > -    install_requires=[
> > -        "pecan",
> > -    ],
> > -    package_data={
> Hm, why was this removed?
> 
> st2api does include some non-source code data (HTML templates).
> 
> —
> Reply to this email directly or view it on GitHub.
>